### DISCUSSION TITLE
Can we get the percentage of alpha channel in PNG?

### DISCUSSION THREAD

**Discussion by keto33** [ImageMagick/ImageMagick#2107] [Help] [answered]
I look for a fast method (for many files) to get the ratio of visible and transparent parts in PNG files.

**Comment by GeeMack** [comment, marked as answer]
With ImageMagick v6 try this...

     convert image.png -resize 1x1 -format "%[fx:a]\n" info:

That should give you the amount of alpha as a decimal, with 0 meaning the alpha channel is fully transparent, and 1 meaning it's fully opaque. Work a command like that into a shell script to run a whole directory at a time.

Note: You can process multiple files at once by running that command with a wildcard "*.png" for input instead of a single image. The output will be each file's alpha ratio printed to the screen on a separate line. The catch: IM will try to read in all the files before doing any processing or outputting, so if the images are large, that's not practical for large quantities.

Use "magick" instead of "convert" for IMv7.

**Reply by Km-1421** [reply]
convert image.png -resize 1x1 -format "%[fx:a]\n" info:

**Comment by fmw42** [comment]
Due to changes between IM 6 and IM 7, these commands give the percentage of transparency.

`convert logo: -transparent white logot.png`

IM 6
`convert logot.png -format "%[fx:100*mean.a]\n" info:`
83.4128

IM 7
`magick logot.png -format "%[fx:100*(1-mean.a)]\n" info:`
83.4128
